tests/validators/typedArray.test.ts

```diff
@@ -0,0 +1,459 @@
+import { ExpectedConstraintError, s, ValidationError } from '../../src';
+import { expectClonedValidator } from '../common/macros/comparators';
+
+describe('TypedArray', () => {
+	describe('Any type of typed array', () => {
+		const predicate = s.typedArray();
+
+		test.each([
+			new Int8Array(),
+			new Uint8Array(),
+			new Uint8ClampedArray(),
+			new Int16Array(),
+			new Uint16Array(),
+			new Int32Array(),
+			new Uint32Array(),
+			new Float32Array(),
+			new Float64Array(),
+			new BigInt64Array(),
+			new BigUint64Array()
+		])('GIVEN %p THEN return the input', (input) => {
+			expect(predicate.parse(input)).toBe(input);
+		});
+
+		test.each([1, true, 'sapphire'])('GIVEN %p THEN throw', (input) => {
+			expect(() => predicate.parse(input)).toThrow(new ValidationError('s.typedArray', `Expected a TypedArray`, input));
+		});
+	});
+
+	describe('i8', () => {
+		const predicate = s.i8;
+
+		test('GIVEN typed array THEN return the input', () => {
+			const typedArray = new Int8Array();
+			expect(predicate.parse(typedArray)).toBe(typedArray);
+		});
+
+		test.each([1, true, 'sapphire', new Int16Array()])('GIVEN %p THEN throw', (input) => {
+			expect(() => predicate.parse(input)).toThrow(new ValidationError('s.typedArray', `Expected a i8 array`, input));
```

Comment:
"a i8"?

Also, I'm not happy with `i8 array`, it should be named after the classes, e.g. `Expected an Int8Array`.